### ISSUE TITLE
m_chanfilter extban

### ISSUE THREAD

**Issue by voidzero** [inspircd/inspircd#1173]
I'd like to be able to set filters per user (or registered account). This might be convenient for example when a user has some kind of auto away message, or for other kinds of repetitive behaviour. I was thinking of something like 

`/mode #channel +b g:*some*filtered*word*:R:acctname`

and

`/mode #channel +b g:*auto?away*after*:*!*lamest@dyn*.example.com`

I guess this conforms somewhat to the other extbans, syntax-wise: first the what, then the who. Other ideas are welcome too.

**Comment by GLolol** [comment]
I think the `[what]` and `[who]` positions may need to be swapped, and possibly using a different separator. It's not always clear which part of the ban is the text to match, and which part is the glob. One could be filtering text or banning a hostname with literal colons in them (e.g. `+b *some:bad:string*:*!*@some:ipv6:ip` can be ambiguous and difficult to parse).

**Comment by voidzero** [comment]
@GLolol, yes, I thought the same thing initially, but realised that would make it inconsistent with the other extbans that have what before who. But yes, fair point regarding colons.

But... I thought of another solution: make m_chanfilter a thing that solely uses extbans. In that case, you would have

`/mode #channel +b g:[who]`

and to make it channel wide, all one has to do would be:

`/mode #channel +b g:*!*@*`

This might be the most simple option.
